### PR TITLE
feat(CommunitiesPortal): Implement text search using mocked model 

### LINKED ISSUE #6360: Search doesn't work on the Communities Portal page
# Bug Report

## Steps to reproduce
1. Create a few communities
2. Open Communities Portal and click on the Search field

#### Expected behavior
The cursor appears in the search field and it's possible to search a community

#### Actual behavior
The cursor doesn't appear in the search field and it's impossible to search a community
### Additional Information
- Status desktop version: https://ci.status.im/job/status-desktop/job/branches/job/macos/job/master/2275/
- Operating System: macOS Monterey, 12.3 Beta (21E5227a)

### LINKED ISSUE #7299: [CommunityPortal] implement designs when 0 search results found
### Description

Implement designs for empty state after tab search shows no results

https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=2563%3A538552

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

Implements scrollable grid view of communities with search by text and some adjustments with design. Data come from dummy model because backend is not ready yet.

Fixes #6360 
Fixes #7299 

### Affected areas

CommunitiesPortal

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

[Kazam_screencast_00018.webm](https://user-images.githubusercontent.com/20650004/192873249-1b6deb5b-e171-4629-9dad-767c2cc07920.webm)

### CODE REVIEW

File: ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml [hunk 1]
```diff
@@ -34,164 +35,247 @@ StatusSectionLayout {
...
                         }
                     }
                 }
             }
         }
+
+        StatusBaseText {
+            anchors.centerIn: parent
+            visible: d.searchMode && searchModel.count === 0
+            text: "No communities found"
```
Comment (caybro) [inline ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml:276]: `qsTr()`

**Review by caybro — COMMENTED** [review]
Looks good!

File: ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml [hunk 2]
```diff
@@ -34,164 +35,247 @@ StatusSectionLayout {
...
         function navigateToCommunity(communityId) {
             root.communitiesStore.setActiveCommunity(communityId)
         }
+
+        readonly property bool searchMode: searcher.text.length > 0
+    }
+
+    // TODO: should be replaced with root.communitiesStore.curatedCommunitiesModel
+    // when backend is ready
+    CommunitiesPortalDummyModel {
```
Comment (alexandraB99) [inline ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml:60]: Could you move this in store in a curatedCommunitiesModel property and use it here through the store instead? (I'd expect also the .qml file to be inside the "stores" folder)

**Review by alexandraB99 — CHANGES_REQUESTED** [review]

File: ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml [hunk 3]
```diff
@@ -34,164 +35,241 @@ StatusSectionLayout {
...
 
     centerPanel: Item {
-        implicitWidth: parent.width
-        implicitHeight: parent.height
+        anchors.top: parent.top
+        anchors.bottom: parent.bottom
         anchors.left: parent.left
+        width: d.layoutWidth
+
+        anchors.topMargin: d.layoutTopMargin
```
Comment (alexandraB99) [inline ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml:100]: could you please check that all content is aligned with section header before adding top margins? this job was done [here](https://github.com/status-im/status-desktop/pull/7342/files#diff-4d4f432ed9442e3f1baa07349d2da11d741b4aeea46523f4ff28a06b307ef190) so please make sure no regression is introduced

File: ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml [hunk 4]
```diff
@@ -34,164 +35,241 @@ StatusSectionLayout {
...
+                            id: featuredLabel
+                            visible: !d.searchMode
+                            Layout.topMargin: d.scrollViewTopMargin
+                            text: qsTr("Featured")
+                            font.weight: Font.Bold
+                            font.pixelSize: d.subtitlePixelSize
+                            color: Theme.palette.directColor1
+                        }
+
+                        GridLayout {
```
Comment (alexandraB99) [inline ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml:204]: this file is becoming overwhelming with more than 300LOC, could you try breaking it in smaller panels/views?

**Review by alexandraB99 — CHANGES_REQUESTED** [review]

File: ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml (hunk 3 above)
Comment (micieslak) [reply ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml:100]: I'm matching margins from this design: https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=8159%3A415655. So I presume that it's ok.

File: ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml (hunk 4 above)
Comment (micieslak) [reply ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml]: good point, split into two files

File: ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml [hunk 5]
```diff
@@ -33,165 +35,139 @@ StatusSectionLayout {
...
 
+    SortFilterProxyModel {
+        id: searchModel
+
+        sourceModel: root.communitiesStore.curatedCommunitiesModel
+
+        filters: ExpressionFilter {
+            enabled: d.searchMode
+            expression: {
+                searcher.text
```
Comment (caybro) [inline ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml:61]: Reevaluation?

**Review by caybro — APPROVED** [review]
Looks good!

File: ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml (hunk 4 above)
Comment (caybro) [reply ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml]: I smiled at 300 == overwhelming :)

File: ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml (hunk 5 above)
Comment (micieslak) [reply ui/app/AppLayouts/CommunitiesPortal/CommunitiesPortalLayout.qml:61]: Yes, `ExpressionFilter` detects dependencies in a specific way and it causes that tricks like that are needed.

**Review by elina2015 — CHANGES_REQUESTED** [review]
@micieslak there are some fixes needed:
 **For the empty state:**
1. The font size should be bigger
2. The space between the tags and the text should match the Design
image
Could you kindly check?
**For the search field:** 
There is no button for deleting the search text
image

**Review by noeliaSD — COMMENTED** [review]
One small issue:
- I think the searcher input should be focused by default when the user enters into this page.

Question:
- What do you mean by "Data come from dummy model because backend is not ready yet." ? Did something changed for not having the backend working now? It was already done by @richard-ramos afaik.

**Comment by micieslak** [comment]
@elina2015 Thanks for your input! I think it's already fixed :)

**Review by elina2015 — APPROVED** [review]
Tested and approved

**Pull request merged**